event_handlers/rust_events.rb

```diff
@@ -4,50 +4,57 @@
 require './helpers/rust_helpers'
 
 # Object that handles events coming from the Rust server
-class RustEvents
-  def initialize(connection_factory, helpers)
-    EM.run do
-      @ws = connection_factory.wrcon_connection
-      @helpers = helpers
-      @rust_helpers = RustHelpers.new(@helpers)
-      @rust_channel = @helpers.get_discord_channel('rust-server')
-      open
-      message
-      close
-      error
+module RustEvents
+  extend Discordrb::EventContainer
```

Comment:
If I'm understanding it correctly - this creates a hard dependency on Discord meaning if Discord goes down or the connection is lost, the Rust portion of the bot no longer functions.

Reply:
This just allows this module to be included into the bot's internal event container.
The connection should still persist, because I'm using memoization on thread creation.
(`@rust_eventmaching ||= Thread.new`)